azurerm/internal/services/network/firewall_policy_rule_collection_group_resource.go

```diff
@@ -0,0 +1,816 @@
+package network
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-05-01/network"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/locks"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmFirewallPolicyRuleCollectionGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmFirewallPolicyRuleCollectionGroupCreateUpdate,
+		Read:   resourceArmFirewallPolicyRuleCollectionGroupRead,
+		Update: resourceArmFirewallPolicyRuleCollectionGroupCreateUpdate,
+		Delete: resourceArmFirewallPolicyRuleCollectionGroupDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.FirewallPolicyRuleCollectionGroupID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FirewallPolicyRuleCollectionGroupName(),
+			},
+
+			"firewall_policy_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FirewallPolicyID,
+			},
+
+			"priority": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(100, 65000),
+			},
+
+			"application_rule_collection": {
```

Comment:
This block has an internal block called `rule`, which is a set of rules. Also this block has its own "name", "priority" and "action", which might be better kept as is so that we can reasonably name those properties as "collection name", "collection priority", etc.

Also, this might be a less burden for users who are used to `azurerm_firewall_applicatioon_rule_collection` resource, to migrate to using this new resource.